### PR TITLE
Removed Picasso to be replaced by Glide.

### LINKED ISSUE #403: Unify the image processor libraries
At the moment we use both [Picasso](https://square.github.io/picasso/) and [Glide](https://bumptech.github.io/glide/). We don't need multiple image processor libraries though. Both libraries have caching techniques for the loaded images but they don't share this cache.
Using a single library will improve the maintainability and performance. 

Steps to be taken:
1. Decide on the library to use
2. Replace all image loading code with the chosen library
4. Add proper configuration for the library (caching settings and such)
8. Ship it 🚢 

---
I assigned this to 0.13.x but we can already use a single library for all new code in 0.12.x

### PR DESCRIPTION
**Pull request by tukilo**
**Changes**

Migration from Glide version 3 to 4.
Removed completly Picasso and OkHttp from the project to be replaced by Glide.
Created new BackgroundManager util to set the background drawable.
Removed unused imports.

**Issues**

Closes #403

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.leanback.app.BackgroundManager
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
+import com.bumptech.glide.load.resource.bitmap.CenterInside
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+
+/** Created by Oriol on 30/05/20. **/
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:14]: We don't add "created by" comments to files. We only have them in some other files that were inherited from the old Emby code.

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+fun BackgroundManager.drawable(
+	activity: Activity,
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:17]: This could be a context

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		.override(widthPixels, heightPixels)
+		.transform(transformation)
+		.listener(object : RequestListener<Drawable?> {
+			override fun onLoadFailed(e: GlideException?, model: Any, target: Target<Drawable?>, isFirstResource: Boolean): Boolean {
+				return false
+			}
+
+			override fun onResourceReady(resource: Drawable?, model: Any, target: Target<Drawable?>, dataSource: DataSource, isFirstResource: Boolean): Boolean {
+				drawable = resource
+				return false
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:37]: We should return `true` here

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt (hunk 2 above)
Comment (tukilo) [reply app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:17]: indeed, what a mistake

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+fun BackgroundManager.drawable(
+	context: Context,
+	url: String,
+	widthPixels: Int,
+	heightPixels: Int
+) {
+
+	Glide.with(context)
+		.load(url)
+		.override(widthPixels, heightPixels)
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:22]:
```suggestion
		.override(widthPixels, heightPixels)
		.centerCrop()
```

When testing I noticed some backdrops getting stretched which did not happen before this PR. Adding the centerCrop function here fixed it.

**Comment by nielsvanvelzen** [comment]
I applied the "blocked" tag because our own CI and Codacy are not working at the moment. The PR is approved unless the pipeline fails or Codacy reports errors once working again.

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt (hunk 4 above)
Comment (tukilo) [reply app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:22]: And it makes sense because I eliminated it without realizing

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import android.graphics.drawable.Drawable
+import androidx.leanback.app.BackgroundManager
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import timber.log.Timber
+
+fun BackgroundManager.drawable(
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:13]: As this file only contains a single extension function I'd recommend renaming it to "BackgroundManagerExtensions" to stay consistent with the other file(s)

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	context: Context,
+	url: String,
+	widthPixels: Int,
+	heightPixels: Int
+) {
+
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:19]: Unnecessary whitespace

File: app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+				return false
+			}
+
+			override fun onResourceReady(resource: Drawable?, model: Any, target: Target<Drawable?>, dataSource: DataSource, isFirstResource: Boolean): Boolean {
+				drawable = resource
+				return true
+			}
+		}).submit()
+
+
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/BackgroundManagerUtils.kt:36]: Unnecessary whitespace
